src/app/courses/course-editor/course-editor.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit } from '@angular/core';
+import { switchMap } from 'rxjs/operators';
+import { of, Observable } from 'rxjs';
+
+import { Course } from '../course.model';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { CourseService } from '../services/course.service';
+import { CanComponentDeactivate } from 'src/app/core/guards/can-deactivate.guard';
+
+@Component({
+  selector: 'app-course-editor',
+  templateUrl: './course-editor.component.html',
+  styleUrls: ['./course-editor.component.scss']
+})
+export class CourseEditorComponent implements OnInit, CanComponentDeactivate {
+  public newCourse: Course;
+  public editedCourse: Course;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private courseService: CourseService
+  ) { }
+
+  ngOnInit() {
+    this.editedCourse = null;
+    this.newCourse = this.createNewCourse();
```

Comment:
You could write 
this.newCourse = new Course(this.generateCourseId());
according to commit bellow